linux-glowforge: Disable imx bus frequency scaling

Upstream-Status: Inappropriate [embedded specific]

Signed-off-by: Scott Wiederhold <s.e.wiederhold@gmail.com>

diff -u a/arch/arm/mach-imx/busfreq-imx.c b/arch/arm/mach-imx/busfreq-imx.c
--- a/arch/arm/mach-imx/busfreq-imx.c
+++ b/arch/arm/mach-imx/busfreq-imx.c
@@ -1285,7 +1285,7 @@
 	ultra_low_bus_freq_mode = 0;
 	cur_bus_freq_mode = BUS_FREQ_HIGH;

-	bus_freq_scaling_is_active = 1;
+	bus_freq_scaling_is_active = 0;
 	bus_freq_scaling_initialized = 1;

 	ddr_low_rate = LPAPM_CLK;
