--- git/src/hal/os/linux/kernel/platform/freescale/gc_hal_kernel_platform_imx.c
+++ git/src/hal/os/linux/kernel/platform/freescale/gc_hal_kernel_platform_imx.c
@@ -60,6 +60,7 @@
 #include "gc_hal_kernel_device.h"
 #include "gc_hal_driver.h"
 #include <linux/slab.h>
+#include <linux/reset.h>
 
 #if defined(CONFIG_PM_OPP)
 #include <linux/pm_opp.h>
@@ -1238,7 +1239,9 @@
     int core = 0;
     struct gpu_clk *imx_clk = NULL;
     struct imx_priv *priv = &imxPriv;
+#ifdef CONFIG_PM
     struct device *pmdev_last = NULL;/*legacy gpu device entry for imx6*/
+#endif
     struct clk *clk_core_last = NULL;/*vg has same core clk as 2d */
 
     for (core = 0; core < gcdMAX_GPU_COUNT; core++) {
